test/runtests.jl

```diff
@@ -8,4 +8,34 @@ using TestItems
     using LinearAlgebra: norm
     using Meshes
     using Unitful
+
+    # environment settings
+    isCI = "CI" ∈ keys(ENV)
+
+    # float settings
+    AD = if isCI
+        if ENV["AD"] == "FiniteDifference"
+            FiniteDifference()
```

Comment:
Currently, one snag with this approach is that several of the geometries with `specialization` methods are edge cases that use `integral` methods with hard-coded `Analytical` differentials. Trying to integrate those geometries with any other `DifferentiationMethod` is guaranteed to throw an error.

This was one of the reasons I was hoping to just eliminate `Analytical` altogether. They're faster for a couple of specific geometry types, but at the cost of a fair bit of added complexity. There was some benefit in code/math reduction, but largely it was to eliminate these cases where `diff_method` isn't truly supported.

Reply:
I'm planning to write up a Discussion post with my thoughts, and some historical context, on our unit test structure/organization. I suspect it might be adequate to just add a single new test to each geometry's standard test set, something like
```julia
@test integral(f, geometry, GaussLegendre(100); diff_method=AutoEnzyme())
```

I think the impact of this new `diff_method` should be apparent for any integral of each geometry type, which would make re-running all of them redundant. And, especially with other recent performance and CI runtime improvements, we currently have a little CI runtime to spare.

Reply:
> Currently, one snag with this approach is that several of the geometries with specialization methods are edge cases that use integral methods with hard-coded Analytical differentials. Trying to integrate those geometries with any other DifferentiationMethod is guaranteed to throw an error.

I left those tests to use `integral`, not the `integral_test` functions I created in `Setup`, this is not an issue.

> This was one of the reasons I was hoping to just eliminate Analytical altogether. They're faster for a couple of specific geometry types, but at the cost of a fair bit of added complexity. There was some benefit in code/math reduction, but largely it was to eliminate these cases where diff_method isn't truly supported.

Personally, I think the added complexity is worth the speed. But that is just my opinion.

> I suspect it might be adequate to just add a single new test to each geometry's standard test set, something like

This makes sense. I will change to this approach.